### ISSUE TITLE
Integrate codecov

### ISSUE THREAD

**Issue by PavelBlend** [taichi-dev/taichi_elements#35]
After extensive refactoring, it is difficult to retest the code. It would be nice to have code coverage information with codecov. And again: I never integrated codecov. @yuanming-hu  Do you have any experience with codecov?

**Comment by yuanming-hu** [comment]
Good point. I think it's good to have code coverage information, but I have no experience with that. I guess maybe we should first have a few unit tests on the UI nodes?

**Comment by PavelBlend** [comment]
I will try to find out about the tests in the blender. At the moment, I have no experience in creating tests.

**Comment by PavelBlend** [comment]
@yuanming-hu at the moment we have `engine` module tests. Here is the code coverage report:
https://codecov.io/github/taichi-dev/taichi_elements/commit/8a919456c07135d031d56c8860e6cb629fb93f0d

Question: Do you know why the `__init__.py` and `mpm_solver.py` files are missing in the report? There is only one `voxelizer.py` file in the report.

How to fix it?

**Comment by PavelBlend** [comment]
I read the build file:
https://codecov.io/codecov/v4/raw/2020-04-21/449790176A7843077D0214BE37F836FB/8a919456c07135d031d56c8860e6cb629fb93f0d/28198800-3523-4563-9975-dd8fea666d67.txt

This file contains information about engine \ mpm_solver.py
But this information is not displayed here:
https://codecov.io/gh/taichi-dev/taichi_elements/tree/8a919456c07135d031d56c8860e6cb629fb93f0d

I do not know the reason. Maybe this is codecov error?

**Comment by PavelBlend** [comment]
I experimented for a long time. All the same, I found the reason for the absence of mpm_solver. The fact is that modules must have unique names. We had two modules in the repository named mpm_solver. I renamed one module and everything worked out. I do not know if this is a codecov error. And I don't know if codecov can be configured to handle files with the same name.

Here is the report:

https://codecov.io/gh/taichi-dev/taichi_elements/tree/90eb5ac241c9254d0dd0d6e8d0d7dacd9ca5eb35/engine
